File: C++/ContainerWithMostWater.cpp
class Solution {
public:
    int maxArea(vector<int> &height) {
        int maxWater = 0;
        int start = 0, end = height.size() - 1;
        while (start < end) {
            int water = (end - start) * min(height[start], height[end]);
            maxWater = max(maxWater, water);
            if (height[start] < height[end])
                ++start;
            else
                --end;
        }
        
        return maxWater;
    }
};

File: javascript/ContainerWithMostWater.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
  let i = 0, j = height.length - 1, maxArea = 0
  while (i < j) {
      maxArea = Math.max(maxArea, Math.min(height[i], height[j]) * (j - i))

      if (height[i] < height[j]) {
          ++i
      } else {
          --j
      }
  }

  return maxArea
};